llpc/tool/llpcCompilationUtils.cpp

```diff
@@ -484,21 +485,18 @@ Error processInputStages(ICompiler *compiler, CompileInfo &compileInfo, ArrayRef
       if ((stageMask & compileInfo.stageMask) != 0)
         break;
 
-      if (stageMask == 0)
+      const auto stages = maskToShaderStages(stageMask);
+      if (stages.empty())
         return createResultError(Result::ErrorInvalidShader,
                                  Twine("Failed to identify shader stages by entry-point \"") + compileInfo.entryTarget +
                                      "\"");
 
-      for (unsigned stage = ShaderStageVertex; stage < ShaderStageCount; ++stage) {
-        if (stageMask & shaderStageToMask(static_cast<ShaderStage>(stage))) {
-          StandaloneCompiler::ShaderModuleData shaderModuleData = {};
-          shaderModuleData.shaderStage = static_cast<ShaderStage>(stage);
-          shaderModuleData.spirvBin = spvBin;
-          compileInfo.shaderModuleDatas.push_back(shaderModuleData);
-          compileInfo.stageMask |= shaderStageToMask(static_cast<ShaderStage>(stage));
-          break;
-        }
-      }
+      ShaderStage stage = stages.front();
```

Comment:
Slightly unrelated to the change but it caught my attention, should we assert that there's only a single stage here?

Reply:
See the if above: `if (stages.empty()) return ...`

Reply:
Yes, that's exactly what happens in the code. We always pick the first one, as ordered by the list of stages in the enum definition (`Vkgc::ShaderStage`). This change preserves the existing logic. Before, the loop would break after finding the first stage.

In this PR, I want to preserve the existing logic and not do any unrelated clean ups if I can avoid them.

Reply:
And an assertion like you suggest fails in LIT tests. There is sometimes more than one stage.